winpr/include/winpr/stream.h

```diff
@@ -50,35 +50,59 @@ WINPR_API BOOL Stream_EnsureRemainingCapacity(wStream* s, size_t size);
 WINPR_API wStream* Stream_New(BYTE* buffer, size_t size);
 WINPR_API void Stream_Free(wStream* s, BOOL bFreeBuffer);
 
-#define _stream_read_n8(_t, _s, _v, _p) do { _v = \
+static INLINE void Stream_Seek(wStream* s, size_t _offset)
+{
+	struct _wStream* _s = (struct _wStream*)s;
+	_s->pointer += (_offset);
+}
+
+static INLINE void Stream_Rewind(wStream* s, size_t _offset)
+{
+	struct _wStream* _s = (struct _wStream*)s;
+	_s->pointer -= (_offset);
+}
+
+#define _stream_read_n8(_t, __s, _v, _p) do { \
+	struct _wStream* _s = (struct _wStream*)__s; \
```

Comment:
@nfedera I've left this in for testing purposes. This way it is quite easy to add ```typedef void* wStream``` and still have the whole thing compile (but complain about direct access to struct members)